library/Director/ProvidedHook/CubeLinks.php

```diff
@@ -19,6 +22,8 @@ public function prepareActionLinks(Cube $cube, View $view)
             return;
         }
 
+        $directorHosts = array_keys(IcingaObject::loadAllByType('Host', $this->directorDb()));
```

Comment:
This is not a good idea, loading all Hosts is not an option. While the user might select just a bunch of them, the DB can easily contain 100k+.

Reply:
That's already the case here: https://github.com/Icinga/icingaweb2-module-director/pull/2436/files#diff-37cfa9169f450b93bda4663c3e82742f0d45336115ae2394ab9c831cab184be9R29, Just FYI

Reply:
@nilmerg: thanks for pointing this out, I overlooked it! @raviks789: please either use the deployment location in IcingaDB as a source of truth or ask the Director about the selected hosts only. If unsure how to address this, please let me know